packages/openactive-integration-tests/test/features/restrictions/booking-window/implemented/opportunity-outside-range-c1-c2-test.js

```diff
@@ -0,0 +1,82 @@
+const chakram = require('chakram');
+const { FeatureHelper } = require('../../../../helpers/feature-helper');
+const { GetMatch, C1, C2 } = require('../../../../shared-behaviours');
+
+/**
+ * @typedef {import('chakram').ChakramResponse} ChakramResponse
+ */
+
+/**
+ * @param {C1|C2} stage
+ * @param {() => ChakramResponse} responseAccessor This is wrapped in a
+ *   function because the actual response won't be available until the
+ *   asynchronous before() block has completed.
+ */
+function itShouldReturn409Conflict(stage, responseAccessor) {
+  it('should return 409', () => {
+    stage.expectResponseReceived();
+    chakram.expect(responseAccessor()).to.have.status(409);
+  });
+}
+
+/**
+ * @param {C1|C2} stage
+ * @param {() => ChakramResponse} responseAccessor This is wrapped in a
+ *   function because the actual response won't be available until the
+ *   asynchronous before() block has completed.
+ */
+function itShouldReturnOpportunityOfferPairNotBookableError(stage, responseAccessor) {
+  it('should return OpportunityOfferPairNotBookableError', () => {
+    stage.expectResponseReceived();
+
+    const response = responseAccessor();
+    const errors = response.body.orderedItem.map(oi => oi.error && oi.error[0] && oi.error[0]['@type']).filter(e => !!e);
+
+    chakram.expect(errors.every(e => e === 'OpportunityOfferPairNotBookableError')).to.equal(true);
```

Comment:
Sorry, this is a weird one, but we try to use chai's `expect` function as much as possible - using chakram's `expect' only for `.to.have.status(..)`, which is useful

chakram is based off of old chai code, which it added some HTTP request stuff to. Therefore, we prefer chai, which is up-to-date and well documented.

(I personally think we should bin chakram at some point for something like axios. It's made for a pre-promise JS world (though it does have promise support now) and is not typed or well documented)